PDAL 1.7.0 fails to build with -Werror=maybe-uninitialized

The Debian package build for PDAL 1.7.0 failed:
```
[ 25%] Building CXX object CMakeFiles/pdal_base.dir/io/LasWriter.cpp.o
/usr/bin/c++  -DUNIX -Dpdal_base_EXPORTS -I/build/pdal-1.7.0/vendor/gtest/include -I/build/pdal-1.7.0/vendor/gtest -I/build/pdal-1.7.0 -I/build/pdal-1.7.0/obj-x86_64-linux-gnu/include -I/build/pdal-1.7.0/vendor -I/build/pdal-1.7.0/vendor/eigen -I/build/pdal-1.7.0/vendor/pdalboost -I/usr/include/gdal -I/usr/include/geotiff -I/usr/include/jsoncpp -I/usr/include/x86_64-linux-gnu -isystem /usr/local/include -I/usr/include/libxml2  -g -O2 -fdebug-prefix-map=/build/pdal-1.7.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -Wdate-time -D_FORTIFY_SOURCE=2 -O2 -g -DNDEBUG -fPIC   -std=c++11 -Wno-implicit-fallthrough -Wno-int-in-bool-context -Wno-dangling-else -Wno-noexcept-type -Wextra -Wpedantic -Werror -Wall -Wno-unused-parameter -Wno-unused-variable -Wpointer-arith -Wcast-align -Wcast-qual -Wredundant-decls -Wno-long-long -Wno-unknown-pragmas -Wno-deprecated-declarations -o CMakeFiles/pdal_base.dir/io/LasWriter.cpp.o -c /build/pdal-1.7.0/io/LasWriter.cpp
In file included from /build/pdal-1.7.0/io/LasWriter.hpp:42:0,
                 from /build/pdal-1.7.0/io/LasWriter.cpp:43:
/build/pdal-1.7.0/io/HeaderVal.hpp: In member function 'virtual void pdal::LasWriter::addArgs(pdal::ProgramArgs&)':
/build/pdal-1.7.0/io/HeaderVal.hpp:42:7: error: '<anonymous>.pdal::BaseHeaderVal<unsigned char>::m_val' may be used uninitialized in this function [-Werror=maybe-uninitialized]
 class BaseHeaderVal
       ^~~~~~~~~~~~~
/build/pdal-1.7.0/io/HeaderVal.hpp:42:7: error: '<anonymous>.pdal::BaseHeaderVal<unsigned char>::m_val' may be used uninitialized in this function [-Werror=maybe-uninitialized]
 class BaseHeaderVal
       ^~~~~~~~~~~~~
/build/pdal-1.7.0/io/HeaderVal.hpp:42:7: error: '<anonymous>.pdal::BaseHeaderVal<short unsigned int>::m_val' may be used uninitialized in this function [-Werror=maybe-uninitialized]
 class BaseHeaderVal
       ^~~~~~~~~~~~~
/build/pdal-1.7.0/io/HeaderVal.hpp:42:7: error: '<anonymous>.pdal::BaseHeaderVal<short unsigned int>::m_val' may be used uninitialized in this function [-Werror=maybe-uninitialized]
 class BaseHeaderVal
       ^~~~~~~~~~~~~
/build/pdal-1.7.0/io/HeaderVal.hpp:42:7: error: '<anonymous>.pdal::BaseHeaderVal<short unsigned int>::m_val' may be used uninitialized in this function [-Werror=maybe-uninitialized]
 class BaseHeaderVal
       ^~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
```

--- a/io/HeaderVal.hpp
+++ b/io/HeaderVal.hpp
@@ -49,7 +49,7 @@ protected:
     BaseHeaderVal() : m_val(T()), m_valSet(false)
     {}
 
-    BaseHeaderVal(const T& t) : m_defVal(t), m_valSet(false)
+    BaseHeaderVal(const T& t) : m_val(T()), m_defVal(t), m_valSet(false)
     {}
 
 public:

